web/packages/teleport/src/Roles/RoleEditor/StandardEditor/standardmodel.ts

```diff
@@ -773,28 +817,56 @@ function stringsToOptions<T extends string>(arr: T[]): Option<T>[] {
 
 function kubernetesResourcesToModel(
   resources: KubernetesResource[] | undefined,
-  roleVersion: RoleVersion
-): { model: KubernetesResourceModel[]; requiresReset: boolean } {
-  const result = (resources ?? []).map(res =>
-    kubernetesResourceToModel(res, roleVersion)
+  roleVersion: RoleVersion,
+  pathPrefix: string
+): {
+  model: KubernetesResourceModel[];
+  conversionErrors: ConversionError[];
+} {
+  const result = (resources ?? []).map((res, i) =>
+    kubernetesResourceToModel(res, roleVersion, `${pathPrefix}[${i}]`)
   );
   return {
     model: result.map(r => r.model).filter(m => m !== undefined),
-    requiresReset: result.some(r => r.requiresReset),
+    conversionErrors: result.flatMap(r => r.conversionErrors),
   };
 }
 
 function kubernetesResourceToModel(
   res: KubernetesResource,
-  roleVersion: RoleVersion
+  roleVersion: RoleVersion,
+  pathPrefix: string
 ): {
   model?: KubernetesResourceModel;
-  requiresReset: boolean;
+  conversionErrors: ConversionError[];
 } {
   const { kind, name, namespace = '', verbs = [], ...unsupported } = res;
+  const conversionErrors = unsupportedFieldErrorsFromObject(
+    pathPrefix,
+    unsupported
+  );
+
   const kindOption = kubernetesResourceKindOptionsMap.get(kind);
+  if (kindOption === undefined) {
+    conversionErrors.push({
+      type: ConversionErrorType.UnsupportedValue,
+      path: pathPrefix,
+    });
+  }
+
   const verbOptions = verbs.map(verb => kubernetesVerbOptionsMap.get(verb));
-  const knownVerbOptions = verbOptions.filter(v => v !== undefined);
+  const knownVerbOptions: KubernetesVerbOption[] = [];
+  for (let i = 0; i < verbOptions.length; i++) {
```

Comment:
Nit: `Array.prototype.forEach` gives you access both to the element and its index without having to write a for loop by hand.